utils/sysUtils/httpclient.go

```diff
@@ -2,10 +2,48 @@
 package sysUtils
 
 import (
+	"context"
 	"net/http"
+	"sync"
+	"time"
 )
 
 // HTTPClient interface
 type HTTPClientI interface {
 	Do(req *http.Request) (*http.Response, error)
 }
+
+type RecycledHTTPClient struct {
+	client *http.Client
+	lock   sync.Mutex
+}
+
+func NewRecycledHTTPClient(ctx context.Context, clientFunc func() *http.Client, ttl time.Duration) *RecycledHTTPClient {
+	recycledHTTPClient := &RecycledHTTPClient{
+		client: clientFunc(),
+	}
+
+	go func() {
+		ticker := time.NewTicker(ttl)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				recycledHTTPClient.lock.Lock()
+				recycledHTTPClient.client = clientFunc()
+				recycledHTTPClient.lock.Unlock()
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return recycledHTTPClient
+}
+
+func (r *RecycledHTTPClient) GetClient() *http.Client {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	return r.client
```

Comment:
There is simpler way to do this, that doesn't require a background go routine.

During a GetClient request we check the timestamp, of ttl is expired we recreate a new one and return it.